android-kit-base/src/main/java/com/mparticle/kits/KitConfiguration.java

```diff
@@ -114,16 +114,200 @@ public SparseBooleanArray getScreenAttributeFilters() {
+        JSONObject consent = new JSONObject();
+        consent.put(KEY_CONSENT_FORWARDING_RULES_SHOULD_INCLUDE_MATCHES, this.consentForwardingIncludeMatches);
+        JSONArray consentArray = new JSONArray();
+        for (Map.Entry<Integer, Boolean> entry : this.mConsentForwardingRules.entrySet()) {
+            JSONObject consentObj = new JSONObject();
+            consentObj.put(KEY_CONSENT_FORWARDING_RULES_VALUE_HASH, entry.getKey());
+            consentObj.put(KEY_CONSENT_FORWARDING_RULES_VALUE_CONSENTED, entry.getValue());
+            consentArray.put(consentObj);
+        }
+        consent.put(KEY_CONSENT_FORWARDING_RULES_ARRAY, consentArray);
+        return consent;
+    }
+
+    private JSONObject getAttributeValueFiltering() throws JSONException {
+        JSONObject filtering = new JSONObject();
+        filtering.put(KEY_ATTRIBUTE_VALUE_FILTERING_SHOULD_INCLUDE_MATCHES, this.avfShouldIncludeMatches);
+        filtering.put(KEY_ATTRIBUTE_VALUE_FILTERING_ATTRIBUTE, this.avfHashedAttribute);
+        filtering.put(KEY_ATTRIBUTE_VALUE_FILTERING_VALUE, this.avfHashedValue);
+        return filtering;
+    }
+
+    public  JSONObject convertToJsonObject() {
+        JSONObject object = new JSONObject();
+        try {
+            object.put(KEY_ID, this.kitId);
+
+            if (this.avfIsActive) {
+                JSONObject filtering = getAttributeValueFiltering();
+                object.put(KEY_ATTRIBUTE_VALUE_FILTERING, filtering);
+            }
+
+            if (!this.settings.isEmpty()) {
+                object.put(KEY_PROPERTIES, this.settings.toString());
+            }
+
+            JSONObject keyFilters = getKeyFilters();
+            if (keyFilters.length() > 0) {
+                object.put(KEY_FILTERS, keyFilters);
+            }
+
+            JSONObject bracketing = getBracketing();
+            if (bracketing.length() > 0) {
+                object.put(KEY_BRACKETING, bracketing);
+            }
+
+            JSONArray customMapping = getCustomMapping();
+            if (customMapping.length() > 0) {
+                object.put(KEY_PROJECTIONS, customMapping);
+            }
+
+            if (this.mConsentForwardingRules != null && !this.mConsentForwardingRules.isEmpty()) {
+                JSONObject consent = getConsentRules();
+                object.put(KEY_CONSENT_FORWARDING_RULES, consent);
+            }
+
+            object.put(KEY_EXCLUDE_ANONYMOUS_USERS, this.mExcludeAnonymousUsers);
+        } catch (JSONException jse) {
+            Logger.error("Issue while converting KitConfigurationToJsonObject: ");
+        }
+        return object;
+    }
+
+    private static JSONObject convertSparseMapToJsonObject(Map<Integer, String> map) throws JSONException {
+        if (map == null || map.size() == 0) {
+            return null;
+        }
+        JSONObject obj = new JSONObject();
+        for (Map.Entry<Integer, String> entry : map.entrySet()) {
+//            try {
```

Comment:
Originally the idea was: because the MParticleSideloadedFilters were not implemented, this will never throw an exception yet, but by the time of implementing it I didnt want to catch Exception without knowing that something happened, so that's why I added the try/catch and commented as if it was a "TODO" comment, but I can change it for a TODO

Reply:
Yeah as a rule we don't check in commented out code. A todo comment is ok as long as you remember to get back to it when finishing the work.